### ISSUE TITLE
error when trying to process GEDCOM file with python-gedcom v. 0.2.0.dev

### ISSUE THREAD

**Issue by 62mkv** [nickreynke/python-gedcom#3]
I've download and installed `python-gedcom` v.0.2.0.dev

I run it as follows: 

```
from gedcom import Gedcom

file_path = '7q4425_661384sh82b72570424am5.ged' # Path to your `.ged` file
gedcom = Gedcom(file_path)

print(gedcom.element_list())
```

This GEDCOM file starts with 
```
0 HEAD
1 GEDC
2 VERS 5.5.1
2 FORM LINEAGE-LINKED
```

and I get the following error:
```
Traceback (most recent call last):
  File "script.py", line 4, in <module>
    gedcom = Gedcom(file_path)
  File "C:\Python3\lib\site-packages\gedcom\__init__.py", line 148, in __init__
    self.__parse(file_path)
  File "C:\Python3\lib\site-packages\gedcom\__init__.py", line 224, in __parse
    last_element = self.__parse_line(line_number, line.decode('utf-8'), last_element)
  File "C:\Python3\lib\site-packages\gedcom\__init__.py", line 263, in __parse_line
    raise SyntaxError(error_message)
SyntaxError: Line `1` of document violates GEDCOM format
See: http://homepages.rootsweb.ancestry.com/~pmcbride/gedcom/55gctoc.htm
```

What am I doing wrong? This GEDCOM file has been exported from MyHeritage recently

UPD: this is with Python 3.6 under Windows 10 x64

**Comment by KeithPetro** [comment]
What happens if you use `gedcom.get_element_list()` in place of `gedcom.element_list()`?

**Comment by 62mkv** [comment]
@keithpetro by looking at the stack trace, one can see that the script does not even reach that statement, so I'm pretty sure it will change nothing

Regarding the format, FWIW the first line is identical, so it shouldn't fail there

**Comment by 62mkv** [comment]
@nickreynke maybe you have an example of Gedcom file that this version can parse successfully? Can you share it? I would analyze the difference and possibly could adapt my file somehow, or update the code

**Comment by KeithPetro** [comment]
I've done a bit more testing and found that the first line in a regular GEDCOM file (like the one I have from Ancestry) should be simply (in byte representation):

```
b'0 HEAD\n'
```

Whereas the file from MyHeritage has:

```
b'\xef\xbb\xbf0 HEAD\r\n'
```

0xEFBBBF is the [BOM (Byte Order Mark) for UTF-8](https://en.wikipedia.org/wiki/Byte_order_mark#UTF-8). This is outside of GEDCOM spec, and I expect that any programs which are able to read these files have to specifically implement out of spec workarounds specifically for MyHeritage files.

**Comment by 62mkv** [comment]
Indeed, the .ged file from MyHeritage has BOM

Thanks!! Now it finally begins to parse. I see that either MyHeritage is shitty on formats, either it is allowed in 5.5.1, but there're multiline entries in exported .ged file, which breaks the parser (Line 32 of document violates GEDCOM format)

**Comment by 62mkv** [comment]
MyHeritage export is ridiculous !! It even splits unicode words in halves!!  so that first byte is at the end of line N, and the second one in the beginning of line N+1

**Comment by KeithPetro** [comment]
@nickreynke I have prepared a (very simple) fix for this. All that's required to ignore BOM at the start of a UTF-8 encoded file is to decode with `'utf-8-sig'` instead of `'utf-8'`.

**Comment by KeithPetro** [comment]
@62mkv I haven't experienced that issue. How are you testing that?

**Comment by 62mkv** [comment]
for some reason, Ancestry.com was able to import MyHeritage GEDCOM file without visible defects...

@KeithPetro what do you mean with "how am I testing that" ?

**Comment by KeithPetro** [comment]
@62mkv What are you using that is showing you that the words are split?

Ancestry's GEDCOM reading code is likely quite robust and allows for various different variations (both valid and invalid) in GEDCOM files.

**Comment by 62mkv** [comment]
Like this one: 
```
2 NOTE Назвали сына Кирюшей дружно и с надеждой, что потом появится Фёдор Кириллович....  В поколении сына имя его был�
3 CONC � крайне редким и сам он любил имя дружка своего Костя В 2007 у них и появится Костя  И в это время имя Кирилла ста
```
Cyrillic is weird but not THAT weird )) Those strange icons are just parts of Unicode word split on different lines. If I remove the CRLF *and 3 CONC* item, it turns into 

```
2 NOTE Назвали сына Кирюшей дружно и с надеждой, что потом появится Фёдор Кириллович....  В поколении сына имя его было крайне редким и сам он любил имя дружка своего Костя В 2007 у них и появится Костя  И в это время имя Кирилла ста
```
As you could notice, the two non-sensical chars have gone and normal Cyrillic letter 'о' has appeared instead of those

**Comment by 62mkv** [comment]
(by "Unicode word" I mean a multi-byte Unicode sequence, describing single character; for Cyrillic it's two bytes)

**Comment by KeithPetro** [comment]
While weird, I think it is actually valid.

According to the GEDCOM standard, the `CONC` tag is meant to signify concatenation without saving the EOL characters prior to the line terminator.

You could split a UTF-16 character mid-way and still properly concatenate it just fine.

**Comment by 62mkv** [comment]
Then it's again an issue with parser, because it stops on all such lines ("byte .. is not a valid utf-8 character")

**Comment by nickreynke** [comment]
@62mkv and @KeithPetro the bug should be resolved by the current release `v0.2.2dev`. ✌
